### PR TITLE
 reorg: modular file reorganization of C source code - v9

### PR DESCRIPTION
**Pull request by jasonish**
Continuation of  https://github.com/OISF/suricata/pull/9960.

This is the same is #9960, but fixes the util migration as a file went missing. Also conforms to CI's idea of clang-format.

Some tests still fail:
- The template tests, fixing the template generation will need to be done manually, so will wait til we're 100% on the layout
- Example plugin with standalone makefile as devel headers aren't installed correctly. This will probably have to be a manual update to the Makefile(s)
- Somethings else are bound to fail.

Next steps:
- [x] `app-layer/`
- [x] `output/eve`
- [x] `output/`
- [x] `util/`
- [x] `util/mpm/`
- [ ] `sources/` (or capture?)
- [ ] `detect/`

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -120,9 +120,9 @@ noinst_HEADERS = \
 	detect-depth.h \
 	detect-detection-filter.h \
 	detect-distance.h \
-	detect-dnp3.h \
-	detect-dns-opcode.h \
-	detect-dns-query.h \
+	app-layer/dnp3/detect.h \
+	app-layer/dns/detect-opcode.h \
+	app-layer/dns/detect-query.h \
```
Comment (jasonish) [inline src/Makefile.am:125]: Will do sort on the file lists when we're good with everything else.

File: src/app-layer-detect-proto.c [hunk 2]
```diff
@@ -2984,135 +2894,55 @@ static int AppLayerProtoDetectTest15(void)
...
-                                  STREAM_TOSERVER,
-                                  ProbingParserDummyForTesting, NULL);
-    AppLayerProtoDetectPPRegister(IPPROTO_TCP,
-                                  "85",
-                                  ALPROTO_FTP,
-                                  7, 15,
-                                  STREAM_TOSERVER,
-                                  ProbingParserDummyForTesting, NULL);
+    AppLayerProtoDetectPPRegister(IPPROTO_TCP, "80", ALPROTO_SMB, 5, 6, STREAM_TOSERVER,
+            ProbingParserDummyForTesting, NULL);
```
Comment (jasonish) [inline src/app-layer-detect-proto.c:2898]: Compressions like this by clang-format are what caused the drop in line count.

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_dbg.
  

Pipeline 16881

**Comment by ct0br0** [comment]
```
HEAD is now at c1bf95532 detect-engine: use ports only after edge case handling
Rebasing (1/36)
Rebasing (2/36)
Rebasing (3/36)
Rebasing (4/36)
Auto-merging src/detect-asn1.c
CONFLICT (content): Merge conflict in src/detect-asn1.c
Auto-merging src/detect-asn1.h
Auto-merging src/detect-engine-build.c
CONFLICT (content): Merge conflict in src/detect-engine-build.c
Auto-merging src/detect-engine-content-inspection.c
Auto-merging src/detect-engine-port.c
Auto-merging src/detect-engine-siggroup.c
Auto-merging src/detect.h
CONFLICT (content): Merge conflict in src/detect.h
error: could not apply 4ddd586ad... reorg: clang-format src/*
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
```

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16888

**Pull request closed without merging**